Explorer/Assets/DCL/Landscape/Worlds/WorldTerrainGenerator.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Diagnostics;
+using DCL.Landscape.Jobs;
+using DCL.Landscape.NoiseGeneration;
+using DCL.Landscape.Settings;
+using DCL.Landscape.Utils;
+using System.Collections.Generic;
+using System.Threading;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+
+namespace DCL.Landscape
+{
+    public class WorldTerrainGenerator
+    {
+        private const string TERRAIN_OBJECT_NAME = "World Generated Terrain";
+        private const float ROOT_VERTICAL_SHIFT = -0.001f; // fix for not clipping with scene (potential) floor
+
+        private readonly int parcelSize;
+
+        private readonly TerrainGenerationData terrainGenData;
+        private readonly NoiseGeneratorCache noiseGenCache = new ();
+
+        private readonly TerrainFactory factory;
+        private readonly TerrainChunkDataGenerator chunkDataGenerator;
+        private readonly TerrainBoundariesGenerator boundariesGenerator;
+
+        private Transform rootGo;
+        private Transform ocean;
+
+        private int maxHeightIndex;
```

Comment:
This field is hanging detached from the flow: maybe it's better to convert it to the argument in the functions flow so it's clear in which order and what dependencies they require?